docs/modules/release-notes/pages/0.27.adoc

```diff
@@ -19,30 +21,329 @@ To get started, follow xref:pkl-cli:index.adoc#installation[Installation].#
+
+|Column 1, row 3
+|Column 2, row 3
+
+|Column 1, row 4
+|Column 2, row 4
+
+|Column 1, row 5
+|Column 2, row 5
+|===
+
+==== `String` to `Number` conversion improvements.
+
+The `String` to `Number` converter methods, for example,  {uri-stdlib-StringToInt}[`String.toInt()`], can now handle underscore separators (https://github.com/apple/pkl/pull/578[#578], https://github.com/apple/pkl/pull/580[#580]).
+
+[source,pkl]
+----
+myNum = "1_000".toInt() // <1>
+----
+<1> Result: `1000`
+
+==== New module: `pkl:analyze`
+
+As part of <<import-analysis>>, a new standard library module is added called `pkl:analyze` (https://github.com/apple/pkl/pull/695[#695]).
+
+This module provides an API for computing the import graph of set of input modules.
+
+Example:
+[source,pkl]
+----
+import "pkl:analyze"
+
+importGraph: analyze.ImportGraph = analyze.importGraph(Set("file:///path/to/my/module.pkl"))
+----
+
+For details, see the {uri-stdlib-analyzeModule}[standard library docs].
+
+==== Support for annotations on a `PklProject` for pkldoc
+
+When creating packages, some annotations on a `PklProject` will affect how pkldoc generates documentation.
+
+* `@Unlisted`: Omit the package from publishing documentation.
+* `@Deprecated`: Add information on the package to show that it is deprecated. 
+
+Example:
+
+.PklProject
+[source,pkl]
+----
+@Deprecated
+amends "pkl:Project"
+----
+
+=== Test report improvements
+
+Tests have been improved 
+
+=== Java API Changes
+
+==== New API: `Analyzer`
 
 == Breaking Changes [small]#💔#
 
 Things to watch out for when upgrading.
 
-=== XXX
+=== Java API Breaking Changes
+
+The following Java APIs have breaking changes:
+
+|===
+|Class/method | Breaking change
+
+|`org.pkl.core.runtime.TestResults`
+|Moved to `org.pkl.core.TestResults`, turn into a record, and restructure.
+
+|===
 
 == Miscellaneous [small]#🐸#
 
-* XXX
+* Make pkl-doc top box include a preview of the rest of module-level documentation (https://github.com/apple/pkl/pull/570[#570]).
+* Make PklProject.deps.json files end in a newline (https://github.com/apple/pkl/pull/664[#644]).
+* Fix invalid syntax of code examples in stdlib doc comments (https://github.com/apple/pkl/pull/703[#703]).
+* Updated Java dependencies (https://github.com/apple/pkl/pull/689[#689]).
 
 == Bugs fixed [small]#🐜#
 
-XXX Bugs down, Inbox Zero in sight ...
+The following bugs have been fixed.
 
-[smaller]
-* XXX (https://github.com/apple/pkl/issues/new[XXX])
+* CLI `--property` flags containing `=` in property values are not parsed correctly (https://github.com/apple/pkl/pull/631[#631]).
+* Thrown `PklBugException` when reading assets from a local project dependency (https://github.com/apple/pkl/pull/642[#642]).
+* Thrown exception when sending `ReadResourceResponse` or `ReadModuleResponse` where both `contents` and `error` are `null` (https://github.com/apple/pkl/pull/657[#657]).
+* Double unary minus is evaluated as single unary minus (https://github.com/apple/pkl/pull/697[#697]).
```

Comment:
Worth calling out handling for evaluating non-ASCII file module URIs and the executor exception handling stuff too?

Reply:
Yup! These release notes are not yet complete.

Reply:
FYI: these release notes are complete now, in case you wanted to review again!